### PR TITLE
Added bucket sort explanation

### PR DESCRIPTION
**Pull request by parulagg27**
**Fixes issue:**

**Changes:**

Added explanation of bucket sort along with it's algorithm & time complexity.

### CODE REVIEW

File: code/sorting/src/bucket_sort/README.md [hunk 1]
```diff
@@ -1,4 +1,36 @@
-# cosmos
-Your personal library of every algorithm and data structure code that you will ever encounter.
+# Bucket Sort
+Bucket sort is a sorting algorithm, that works by distributing the elements of an array into a number of buckets. Then, we apply some sorting algorithm (**insertion sort** for best optimisation ) to sort elements in each bucket, & finally, take the elements out of buckets in order & put them back to original array to get sorted result.
```
Comment (arnavb) [inline code/sorting/src/bucket_sort/README.md]: 1. No need for the first comma.
2. Then, we apply _a_ sorting algorithm
3. **insertion sort** for best performance (Or optimization if you want to keep the word)
4. Better to use and than &
5. get _the_ sorted result.

File: code/sorting/src/bucket_sort/README.md [hunk 2]
```diff
@@ -1,4 +1,36 @@
...
+Elements are first distributed among bins/buckets
+
+![fig2](https://upload.wikimedia.org/wikipedia/commons/e/e3/Bucket_sort_2.svg)
+
+Then, elements are sorted within each bucket.
+
+## Algorithm
+Bucket sort is mainly useful when input is uniformly distributed over a range.
+
+If input is an n element array arr[i] with all elements lying in the range: [0.0, 1.0], & uniformly distributed in the same range, the algorithm is:
```
Comment (arnavb) [inline code/sorting/src/bucket_sort/README.md]: & -> and

Comment (parulagg27) [reply code/sorting/src/bucket_sort/README.md]: Will take care of this from next time. 👍

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**